Kernel/Storage/AHCIController.cpp

```diff
@@ -75,12 +75,14 @@ volatile AHCI::PortRegisters& AHCIController::port(size_t port_number) const
 
 volatile AHCI::HBA& AHCIController::hba() const
 {
-    return static_cast<volatile AHCI::HBA&>(*(volatile AHCI::HBA*)(m_hba_region->vaddr().as_ptr()));
+    // See default_hba_region for the reason this offset is here.
+    return static_cast<volatile AHCI::HBA&>(*(volatile AHCI::HBA*)(m_hba_region->vaddr().as_ptr() + m_hba_offset));
 }
 
 AHCIController::AHCIController(PCI::Address address)
     : StorageController()
     , PCI::Device(address)
+    , m_hba_offset(0)
```

Comment:
Please initialize this in the class declaration instead of here. :)

Reply:
Done :+1:
When I tried it in the class declaration before, it would reset the value I set in default_hba_region, but it doesn't seem to do that anymore (probably the order of the members?)